lib/migrate.js

```diff
@@ -0,0 +1,444 @@
+  };
+
+  /**
+ * Add and delete certain indexes, performs depends on which function calls it.
+ * 
+ * @param {String[]} [models] A model name or an array of model names. Passed in
+ * from `automigrate` or `autoupdate`
+ * @param {boolean} isMigrate 
+ * - `true` when called by `automigrate`
+ * - `false` when called by `autoupdate`
+ * 
+ * @callback {Function} cb The callback function
+ */
+
+  CouchDB.prototype.migrateOrUpdateIndex = function(models, isMigrate, cb) {
+    debug('CouchDB.prototype.migrateOrUpdateIndex %j', models);
+
+    var self = this;
+    async.each(models, autoUpdateOneModel, cb);
+
+    function autoUpdateOneModel(model, cb) {
+      debug('CouchDB.prototype.migrateOrUpdateIndex updating model: %j', model);
+      var mo = self.selectModel(model, true);
+      assert(mo, 'model ' + model + ' does not exist in your registry!');
+      var indexes = mo.mo.model.definition.indexes();
+
+      self.getModifyIndexes(mo, indexes, isMigrate, function(err, results) {
+        debug('start drop and add indexes %j for model %j', results, model);
+        if (err) return cb(err);
+        assert(typeof results === 'object',
+          'results of modified indexes must be an object!');
+        async.series([
+          function dropIndexes(cb) {
+            removeIndexes(results.indexesToDrop, cb);
+          },
+          function addIndexes(cb) {
+            createIndexes(results.indexesToAdd, cb);
+          },
+        ], cb);
+      });
+
+      function createIndexes(indexes, cb) {
+        assert(typeof indexes === 'object', 'indexes to create must be an object!');
+        async.eachOf(indexes, create, cb);
+        // {indexName: "foo"} or {indexName: [{"foo": "asc"}, {"bar": "asc"}]}
+        function create(value, key, cb) {
+          createIndex(key, value, cb);
+        };
+      };
+
+      function createIndex(name, fields, cb) {
+      // naming convertion: '_design/LBModel__Foo__LBIndex__foo_index',
+      // here the driver api takes in the name without prefix '_design/'
+        var config = {
+          ddocName: self.getIndexModelPrefix(mo) + '__' + model + '__' +
+          self.getIndexPropertyPrefix(mo) + '__' + name,
+          indexName: name,
+          fields: fields,
+        };
+        self.createIndex(config.ddocName, config.indexName, config.fields, cb);
+      }
+
+      function removeIndexes(indexes, cb) {
+        assert(typeof indexes === 'object', 'indexes to drop must be an object!');
+        async.eachOf(indexes, removeIndex, cb);
+        // {ddoc: "_design/ddocName", name: "indexName"}
+        function removeIndex(value, key, cb) {
+          self.deleteIndex(mo, value.ddoc, cb);
+        }
+      };
+    };
+  };
+
+  /**
+ * Create an index in couchdb, you can specify the ddocName and indexName
+ * @param {String} ddocName without prefix `_design/`
+ * @param {String} indexName index name
+ * @param {Array} fields example format: [{field1: 'asc'}, {field2: 'asc'}]
+ */
```

Comment:
missing `@callback {Function} cb The callback function`
I feel lots of jsdocs are missing and probably not following the correct annotation....I will do another PR to clean up them.